src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -128,6 +131,13 @@ public SnowflakeSinkServiceV2(
     this.partitionsToChannel = new HashMap<>();
 
     this.tableName2SchemaEvolutionPermission = new HashMap<>();
+
+    // jmx
+    String connectorName =
+        conn == null || Strings.isNullOrEmpty(this.conn.getConnectorName())
+            ? "default_connector"
```

Comment:
Ideally this is not possible since name is a requirement but good to check.

Reply:
My two cents: we should not do this if the assumption is connector name should never be null, you can add an assert instead